tex/polyglossia-punct.lua

```diff
@@ -150,7 +208,21 @@ local function process(head)
                                 head = remove_node(head, next)
                             end
                         end
-                        head, start = insert_node_after(head, start, get_kern_node(rightspace*quad))
+                        if rightspace.unit == "quad" then
+                            unit = fontparameters.quad
+                            spacing_node = get_kern_node(rightspace.kern*unit)
+                        elseif rightspace.unit == "space" then
+                            unit = fontparameters.space
+                            if rightspace.rubber then
+                                stretch = rightspace.kern*fontparameters.space_stretch
+                                shrink  = rightspace.kern*fontparameters.space_shrink
+                                spacing_node = get_glue_node(rightspace.kern*unit, stretch, shrink)
+                                head = insert_node_after(head, start, get_penalty_node())
+                            else
+                                spacing_node = get_kern_node(rightspace.kern*unit)
+                            end
+                        end
+                        head = insert_node_after(head, start, spacing_node)
```

Comment:
We can skip a redundant test of next node, when we insert `start` after `head` like the above comment.

Reply:
You are right. I will correct it.